### ISSUE TITLE
build_info.cpp needs to be created in the build dir, not in the source tree

### ISSUE THREAD

**Issue by ugermann** [marian-nmt/marian-dev#645]
### Bug description

build_info.cpp needs to be created in the build dir, not in the source tree

### How to reproduce

```
mkdir build1 build2
cd build1
cmake  -DCOMPILE_CUDA=off -DUSE_SENTENCEPIECE=on
cd ../build2
cmake  -DUSE_SENTENCEPIECE=off 
cd ../build1
make -j
./marian --build-info
```
That results in USE_SENTENCEPIECE=off, even though the code in build1 was compiled with USE_SENTENCEPIECE=on. -DCOMPILE_CUDA=off in build1 is just to speed up compilation for demonstrating this bug.

**Comment by ugermann** [comment]
Fixed in #644, by the way.

**Comment by snukky** [comment]
PR was merged. Closing. Thanks!
